dxr/app.py

```diff
@@ -114,27 +122,34 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             if is_case_sensitive:
                 params['case'] = 'true'
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
+
     try:
-        count_and_results = query.results(offset, limit)
-        # Convert to dicts for ease of manipulation in JS:
-        results = [{'icon': icon,
-                    'path': path,
-                    'lines': [{'line_number': nb, 'line': l} for nb, l in lines],
-                    'is_binary': is_binary}
-                   for icon, path, lines, is_binary in count_and_results['results']]
+        count, results = query.results(offset, limit)
+        term = query.single_term()
+        promoted_count, promoted = 0, []
+        # Only offer promoted results when at the top of the page and we have only a single term.
+        if offset == 0 and term:
```

Comment:
Actually playing with the UI, I was surprised to find that, when I brought up a context menu on a path result (cool—that works!) and chose "Limit search to folder", I stopped getting path results. I think the condition we really want here isn't "there's a single term" but rather "there's a single text term, and all the rest of the filters are in the FILE domain". That way, the context menu would work, and we could get path results when searching for things like `ext:c main`.